spec/margin.md

```diff
@@ -63,69 +35,137 @@ pub struct MarginAccount {
 }
 ```
 
-## Messages
+```rust
+// Position holds a store of two token denominations (or mints). One for the deonomination that the loan is in
+// and the other for the denomination that the trade or collateral is in.
+pub struct Position {
```

Comment:
What do you guys think about this? Is it somewhat similar to how margin trading will actually be implemented?

Reply:
I think this should be tracked by the client. Off the top of my head I cant see a reason to keep this data on chain. 

Its a bit cumbersome to keep the data off chain, but that is why liquidators would get a premium for their services.

Reply:
My understanding is that when you take out a loan you first store the funds in a vault in the margin account before actually trading it at an AMM or orderbook. This needs to be locked up so that the owner of the margin account can't just hit withdraw and run away with the funds. 

Also when a trader is holding a position there needs to a locked up token account which the funds remain in until the trader is ready to close the position.

Reply:
oh sorry, you are right. In my mind I had this sitting at a higher level but moving it down makes sense.